edit-post/components/header/fixed-toolbar-toggle/index.js

```diff
@@ -1,49 +1,41 @@
 /**
- * External Dependencies
+ * WordPress Dependencies
  */
-import { connect } from 'react-redux';
+import { withSelect, withDispatch } from '@wordpress/data';
 
 /**
  * WordPress Dependencies
  */
 import { __ } from '@wordpress/i18n';
+import { compose } from '@wordpress/element';
 import { MenuItemsGroup, MenuItemsToggle, withInstanceId } from '@wordpress/components';
+import { ifViewportMatches } from '@wordpress/viewport';
 
-/**
- * Internal Dependencies
- */
-import { hasFixedToolbar, isMobile } from '../../../store/selectors';
-import { toggleFeature } from '../../../store/actions';
-
-function FeatureToggle( { onToggle, active, onMobile } ) {
-	if ( onMobile ) {
-		return null;
-	}
+function FeatureToggle( { onToggle, isActive } ) {
 	return (
 		<MenuItemsGroup
 			label={ __( 'Settings' ) }
 			filterName="editPost.MoreMenu.settings"
 		>
 			<MenuItemsToggle
 				label={ __( 'Fix Toolbar to Top' ) }
-				isSelected={ active }
+				isSelected={ isActive }
 				onClick={ onToggle }
 			/>
 		</MenuItemsGroup>
 	);
 }
 
-export default connect(
-	( state ) => ( {
-		active: hasFixedToolbar( state ),
-		onMobile: isMobile( state ),
-	} ),
-	( dispatch, ownProps ) => ( {
+export default compose( [
+	withSelect( ( select ) => ( {
+		isActive: select( 'core/edit-post' ).isFeatureActive( 'fixedToolbar' ),
+	} ) ),
+	withDispatch( ( dispatch, ownProps ) => ( {
 		onToggle() {
-			dispatch( toggleFeature( 'fixedToolbar' ) );
+			dispatch( 'core/edit-post' ).toggleFeature( 'fixedToolbar' );
 			ownProps.onToggle();
 		},
-	} ),
-	undefined,
-	{ storeKey: 'edit-post' }
-)( withInstanceId( FeatureToggle ) );
+	} ) ),
+	ifViewportMatches( 'medium' ),
```

Comment:
Would it make any difference in case `ifViewportMatches( 'medium' )` would be composed as the first item here? All HOCs would have to be applied when the source file loads, but it seems like `withSelect` and `withDispatch` wouldn't have to be rendered in case the viewport doesn't match. It might only make any difference on the initial render if I process the application flow properly.

Reply:
>Would it make any difference in case `ifViewportMatches( 'medium' )` would be composed as the first item here?

I'll test it, but yes, this seems like it could potentially serve as a (small) optimization.